cmd/influx_inspect/buildtsi/buildtsi.go

```diff
@@ -182,40 +183,37 @@ func (cmd *Command) compactSeriesFilePartition(path string) error {
 	const tmpExt = ".tmp"
 
 	fmt.Fprintf(cmd.Stdout, "processing partition for %q\n", path)
-	fis, err := ioutil.ReadDir(path)
+
+	// Open partition so index can recover from entries not in the snapshot.
+	partitionID, err := strconv.Atoi(filepath.Base(path))
 	if err != nil {
-		return err
+		return fmt.Errorf("cannot parse partition id from path: %s", path)
 	}
```

Comment:
Partition needs to be opened because just opening the index file means that unsnapshotted entries are not replayed.